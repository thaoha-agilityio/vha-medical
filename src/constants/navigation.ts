import {
  AppointmentIcon,
  ChemistIcon,
  DashboardIcon,
  DoctorIcon,
  HospitalIcon,
  MedicineIcon,
  MessageIcon,
  SettingIcon,
} from '@/icons';

// Router
import { PRIVATE_ROUTES } from './router';

export const NAVBAR_LINKS = [
  {
    name: 'Dashboard',
    href: PRIVATE_ROUTES.DASHBOARD,
    icon: DashboardIcon,
    isEnable: true,
  },
  {
    name: 'Appointments',
    href: PRIVATE_ROUTES.APPOINTMENTS,
    icon: AppointmentIcon,
    isEnable: true,
  },
  {
    name: 'Chemists',
    href: PRIVATE_ROUTES.CHEMISTS,
    icon: ChemistIcon,
    isEnable: true,
  },
  { name: 'Hospitals', href: '#', icon: HospitalIcon },
  { name: 'Doctors', href: '#', icon: DoctorIcon },
  { name: 'Medicines', href: '#', icon: MedicineIcon },
];

export const SUPPORT_LINKS = [
  { name: 'Messages', href: '#', icon: MessageIcon },
  {
    name: 'Settings',
    href: '#',
    icon: SettingIcon,
  },
];
