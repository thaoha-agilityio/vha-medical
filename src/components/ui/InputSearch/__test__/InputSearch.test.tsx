import {
  render,
  screen,
  fireEvent,
  waitFor,
  act,
} from '@testing-library/react';

import { InputSearch } from '..';
import { useSearchParams } from 'next/navigation';

jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn().mockReturnValue({
    get: jest.fn().mockReturnValue('1'),
  }),
  useRouter: jest.fn().mockReturnValue({
    replace: jest.fn(),
  }),
  usePathname: jest.fn().mockReturnValue(''),
}));

describe('InputSearch', () => {
  jest.useFakeTimers();

  const mockUseSearchParams = useSearchParams as jest.Mock;
  afterEach(() => {
    jest.clearAllTimers();
    jest.clearAllMocks();
  });

  it('Should render SearchInput Component correctly', () => {
    mockUseSearchParams.mockReturnValue({
      get: jest.fn().mockReturnValue(''),
    });

    const { container } = render(
      <InputSearch
        classNames={{
          mainWrapper: 'mock',
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should invoke handleChangeInput and updateSearchParams when typing', async () => {
    mockUseSearchParams.mockReturnValue({
      get: jest.fn().mockReturnValue('1'),
    });

    render(<InputSearch placeholder="Search" />);

    const searchInput: HTMLInputElement = screen.getByPlaceholderText('Search');

    act(() => {
      fireEvent.change(searchInput, { target: { value: 'Name' } });
    });

    jest.advanceTimersByTime(600);

    await waitFor(() => {
      expect(searchInput.value).toBe('Name');
    });
  });

  it('clears invoke handleClear function when clearing the input', async () => {
    render(<InputSearch placeholder="Search..." value="example" />);

    const input: HTMLInputElement = screen.getByPlaceholderText('Search...');
    const clearButton: HTMLButtonElement = screen.getByRole('button'); // Assuming there's a clear button

    // Simulate clearing the input
    fireEvent.click(clearButton);

    await waitFor(() => {
      fireEvent.change(input, { target: { value: '' } });
    });
  });

  it('should clear the search params when the input is empty', async () => {
    render(<InputSearch placeholder="Search" />);

    const searchInput: HTMLInputElement = screen.getByPlaceholderText('Search');

    act(() => {
      fireEvent.change(searchInput, { target: { value: 'Name' } });
    });

    jest.advanceTimersByTime(600);

    await waitFor(() => {
      expect(searchInput.value).toBe('Name');
    });

    act(() => {
      fireEvent.change(searchInput, { target: { value: '' } });
    });

    jest.advanceTimersByTime(600);

    await waitFor(() => {
      expect(searchInput.value).toBe('');
    });
  });
});
