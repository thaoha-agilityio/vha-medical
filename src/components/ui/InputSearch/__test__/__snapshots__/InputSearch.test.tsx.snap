// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSearch Should render SearchInput Component correctly 1`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
    data-filled="true"
    data-filled-within="true"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="flex flex-col h-fit lg:max-w-[400px] text-xs mock"
      data-slot="main-wrapper"
    >
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-10 rounded-medium transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background bg-transparent border-text-foreground !bg-transparent data-[hover=true]:!bg-transparent group-data-[focus=true]:ring-0 group-data-[focus=true]:ring-offset-0 group-data-[focus=true]:!bg-transparent data-[focus=true]:border-secondary-300 group-data-[invalid=true]:border-danger-100 py-4 border h-[52px]"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border"
          data-slot="inner-wrapper"
        >
          <svg
            class="w-5 h-5 ml-4 text-primary-200"
            fill="none"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_1_18701)"
            >
              <path
                d="M9.58341 17.5C13.9557 17.5 17.5001 13.9556 17.5001 9.58335C17.5001 5.2111 13.9557 1.66669 9.58341 1.66669C5.21116 1.66669 1.66675 5.2111 1.66675 9.58335C1.66675 13.9556 5.21116 17.5 9.58341 17.5Z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M18.3334 18.3334L16.6667 16.6667"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </g>
            <defs>
              <clippath
                id="clip0_1_18701"
              >
                <rect
                  fill="white"
                  height="20"
                  width="20"
                />
              </clippath>
            </defs>
          </svg>
          <input
            aria-label=" "
            class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text peer pr-6 rtl:pr-0 rtl:pl-6 group-data-[has-value=true]:text-default-foreground text-primary-100 placeholder:text-primary-300 bg-transparent text-sm mx-2 group-data-[invalid=true]:!text-danger-100"
            data-has-start-content="true"
            data-slot="input"
            id="react-aria-:r0:"
            placeholder=""
            type="text"
            value=""
          />
          <span
            aria-label="clear input"
            class="p-2 -m-2 z-10 hidden absolute end-3 start-auto appearance-none select-none opacity-0 hover:!opacity-100 cursor-pointer active:!opacity-70 rounded-full outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-large peer-data-[filled=true]:opacity-70 peer-data-[filled=true]:block transition-opacity motion-reduce:transition-none"
            data-slot="clear-button"
            role="button"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="1em"
              role="presentation"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
